integration-cli/docker_cli_rmi_test.go

```diff
@@ -268,3 +268,33 @@ func (s *DockerSuite) TestRmiBlank(c *check.C) {
 		c.Fatalf("Expected error message not generated: %s", out)
 	}
 }
+
+func (s *DockerSuite) TestRmiContainerImageNotFound(c *check.C) {
+	// Build 2 image for testing.
+	imageNames := []string{"test1", "test2"}
+	imageIds := make([]string, 2)
+	for i, name := range imageNames {
+		dockerfile := fmt.Sprintf("FROM busybox\nMAINTAINER %s\nRUN echo %s\n", name, name)
+		id, err := buildImage(name, dockerfile, false)
+		c.Assert(err, check.IsNil)
+		imageIds[i] = id
+	}
+
+	// Create a long-running container.
+	_, status := dockerCmd(c, "run", "-d", imageNames[0], "top")
+	c.Assert(status, check.Equals, 0)
+
+	// Create a stopped container, and then force remove its image.
+	_, status = dockerCmd(c, "run", imageNames[1], "true")
+	c.Assert(status, check.Equals, 0)
+	_, status = dockerCmd(c, "rmi", "-f", imageIds[1])
+	c.Assert(status, check.Equals, 0)
+
+	// Try to remove the image of the running container and see if it fails
+	// as expected.
+	out, _, err := runCommandWithOutput(exec.Command(dockerBinary, "rmi", "-f", imageIds[0]))
```

Comment:
@cpuguy83 `dockerCmd` asserts that the command must success, but here the `rmi -f` is expected to fail, so I kept using `runCommandWithOutput`